quarkchain/cluster/subscription.py

```diff
@@ -80,22 +81,13 @@ def response_encoder(sub_id, result):
         }
 
     @staticmethod
-    def syncing_response_encoder(sub_id, running, tip, queue):
-        if not running:
-            return {
-                "jsonrpc": "2.0",
-                "subscription": sub_id,
-                "result": {"syncing": running},
-            }
+    def sync_status_encoder(sub_id, data):
+        running, header_tip, highest_block = data
 
-        return {
-            "jsonrpc": "2.0",
-            "subscription": sub_id,
-            "result": {
-                "syncing": running,
-                "status": {
-                    "startingBlock": tip,
-                    "highestBlock": max(h.height for h, _ in queue),
-                },
-            },
-        }
+        ret = {"jsonrpc": "2.0", "subscription": sub_id, "result": {"syncing": running}}
+        if running:
+            ret["result"]["status"] = {
+                "startingBlock": header_tip,
```

Comment:
from https://ethereum.stackexchange.com/questions/31145/what-does-startingblock-mean-in-eth-syncing

height of header tip should be current block, and I guess starting block should be the minimum height among the queue?

Reply:
ah make sense!